### PR TITLE
added usefull rubocop config. Updated all files with the rules

### PR DESCRIPTION
**Pull request by fanantoxa**
@soulcutter I've added config for rubocop and updated all files. Please merge it if it fit to you.
If it not please let me know which rules you want to on/off and I'll update PR

### CODE REVIEW

File: lib/saxerator/adapters/ox.rb [hunk 1]
```diff
@@ -54,7 +52,7 @@ def text(str)
         reader.characters(str)
       end
 
-      alias :cdata :text
+      alias cdata text
```
Comment (soulcutter) [inline lib/saxerator/adapters/ox.rb]: ha, this one always bites me.

File: lib/saxerator/builder/hash_builder.rb [hunk 2]
```diff
@@ -60,12 +60,13 @@ def add_to_hash_element(hash, name, element)
 
       def block_variable
         return to_s if @text
-        return to_hash if @children.count > 0 || (@attributes.count > 0 && @config.put_attributes_in_hash?)
+        return to_hash if @children.count > 0 ||
+                          (@attributes.count > 0 && @config.put_attributes_in_hash?)
```
Comment (soulcutter) [inline lib/saxerator/builder/hash_builder.rb]: I strongly dislike whitespace to align things into columns. I'd rather extract methods or variables to keep this short enough to not have to wrap. Any cop that promotes that style should be disabled.

File: lib/saxerator/configuration.rb [hunk 3]
```diff
@@ -30,23 +26,23 @@ def generate_key_for(val)
     end
 
     def hash_key_normalizer
-      @hash_key_normalizer ||= lambda { |x| x.to_s }
+      @hash_key_normalizer ||= -> (x) { x.to_s }
```
Comment (soulcutter) [inline lib/saxerator/configuration.rb]: The current build matrix includes Ruby 1.9.3 which does not have stabby lambdas.

Honestly, though, I'm ok dropping 1.9.3 support. It's a dead ruby at this point.

File: lib/saxerator/configuration.rb [hunk 4]
```diff
@@ -69,7 +65,8 @@ def put_attributes_in_hash?
 
     def raise_error_if_using_put_attributes_in_hash_with_xml
       if @output_type != :hash && @put_attributes_in_hash
-        raise ArgumentError.new("put_attributes_in_hash! is only valid when using output_type = :hash (the default)'")
+        raise ArgumentError, "put_attributes_in_hash! is only valid \
+                              when using output_type = :hash (the default)'"
```
Comment (soulcutter) [inline lib/saxerator/configuration.rb]: Prefer not to indent into columns here.

File: lib/saxerator/latches/child_of.rb [hunk 5]
```diff
@@ -6,20 +6,15 @@ def initialize(name)
...
 
-      def start_element name, _
-        if depth_within_element > 0
-          increment_depth(1)
-        end
-        if @name == name
-          @depths.push 1
-        end
+      def start_element(name, _)
+        increment_depth(1) if depth_within_element > 0
```
Comment (soulcutter) [inline lib/saxerator/latches/child_of.rb]: this looks a lot better this way :+1:

File: spec/spec_helper.rb [hunk 6]
```diff
@@ -1,30 +1,33 @@
...
+  config.example_status_persistence_file_path = 'spec/examples.txt'
   config.include FixtureFile
 
-  config.before(:suite) do |example|
-    return unless adapter = ENV["SAXERATOR_ADAPTER"]
+  adapter = ENV['SAXERATOR_ADAPTER']
+
+  config.before(:suite) do |_|
+    unless adapter
+      puts 'SAXERATOR_ADAPTER weren\'t defined'
```
Comment (soulcutter) [inline spec/spec_helper.rb]: I don't think this should exit here, it should just skip this block. I think the `return` here probably should have been a `next`

**Comment by soulcutter** [comment]
This caught some good stuff, both in style and substance.

One cop I would like to disable is the `StringLiterals` cop - policing single vs double quotes is a total waste, even in an automated tool.

File: lib/saxerator/configuration.rb (hunk 4 above)
Comment (fanantoxa) [inline lib/saxerator/configuration.rb]: @soulcutter yes, but it's really hard to see this line.

**Comment by fanantoxa** [comment]
@soulcutter I've updated rubocop config with your additions.
Also I've added to this PR next tasks:
[Drop support ruby 1.9.3](https://github.com/soulcutter/saxerator/issues/45)
[Add a changelog](https://github.com/soulcutter/saxerator/issues/41)

Please check when you find a time

**Comment by soulcutter** [comment]
Looks great!

**Pull request merged**